### PR TITLE
cdn new version 2022 01 01 preview

### PR DESCRIPTION
**Pull request by priyadmurugan**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.

The new APIs that are added are internal APIs and the target for Private preview is Jan/Feb 2022. Please do not merge these APIs as they have not been implemented yet. 
Note: The new changes are done on top of a new version release for CDN (2021-06-01) which is later merged into main after this PR(https://github.com/Azure/azure-rest-api-specs/pull/17092) was created. 

**Please find the preview PR which was reviewed here: https://github.com/Azure/azure-rest-api-specs/pull/17092.** This new PR is added to add the missing changes from version 2021-06-01

  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.

January/February 

  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[x] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by priyadmurugan** [comment]
@mentat9 : This is the new PR created to address the missing changes from the new version(2021-06-01) that was added.

**Comment by mentat9** [comment]
>           "$ref": "./afdx.json#/definitions/AFDStateProperties"

This enum is missing the Canceled value (ref: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#provisioningstate-property](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#provisioningstate-property)). You should add it, but I won't block ARM approval on it. Note: you can add string enum values without an API version change.

---
Refers to: specification/cdn/resource-manager/Microsoft.Cdn/preview/2022-01-01-preview/cdn.json:6748 in 6db7b08. [](commit_id = 6db7b08a9d9e9e91a3f2cef42cc484a385674363, deletion_comment = False)

**Comment by mentat9** [comment]
Left one comment, signing off from ARM perspective. I also cleaned up labels on the previous PR. Please close it if you are finished with it.

File: specification/cdn/resource-manager/Microsoft.Cdn/preview/2022-01-01-preview/cdn.json [hunk 1]
```diff
@@ -0,0 +1,6813 @@
...
+      "allOf": [
+        {
+          "$ref": "./afdx.json#/definitions/AFDStateProperties"
+        }
+      ],
+      "required": [
+        "ServiceType",
+        "SchemaVersion"
+      ],
+      "SchemaVersion":{
```
Comment (jorinmejia) [inline specification/cdn/resource-manager/Microsoft.Cdn/preview/2022-01-01-preview/cdn.json]: Will SchemaVersion support more than one value? If so can it be changed into an enum?

File: specification/cdn/resource-manager/Microsoft.Cdn/preview/2022-01-01-preview/cdn.json [hunk 2]
```diff
@@ -0,0 +1,6813 @@
...
+        "description": "Schema version of the policy",
+        "type": "string"
+      },
+      "ServiceType": {
```
Comment (jorinmejia) [inline specification/cdn/resource-manager/Microsoft.Cdn/preview/2022-01-01-preview/cdn.json]: Will ServiceType support more than one value? If so can it be changed into an enum?

File: specification/cdn/resource-manager/Microsoft.Cdn/preview/2022-01-01-preview/cdn.json [hunk 3]
```diff
@@ -0,0 +1,6813 @@
...
+          "default": {
+            "description": "CDN error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cdn/profiles/{profileName}/policies/{policyName}": {
```
Comment (jorinmejia) [inline specification/cdn/resource-manager/Microsoft.Cdn/preview/2022-01-01-preview/cdn.json]: Can a profile have more than one policy? If so, would it make sense to have a GET where we can retrieve all the policies in a profile? Similar to how we can get all endpoints in a profile.

**Review by jorinmejia — APPROVED** [review]

**Comment by leni-msft** [comment]
@priyadmurugan  pls fix the lint error. And the "API Readiness Check" failed - it seems the apiVersion hasn't been deployed. The PR will not be merged until the service is deployed.

**Comment by leni-msft** [comment]
@tadelesh @qiaozha please help check the go & js sdk breaking changes.

**Comment by priyadmurugan** [comment]
> @priyadmurugan pls fix the lint error. And the "API Readiness Check" failed - it seems the apiVersion hasn't been deployed. The PR will not be merged until the service is deployed.

@leni-msft : The error is related to the resource type "operations". we are not planning on adding the "operations" resource type to the older API version. So can we ignore this error and merge?

**Comment by priyadmurugan** [comment]
> The API Readiness Check is sending a GET operations request to detect if the new apiVersion has been deployed to prod. It's not just for checking operations API. As explained, swagger cannot be merged to main until the API is deployed.

which endpoint / environment is used for the GET calls?

**Comment by leni-msft** [comment]
@priyadmurugan ` GET https://management.azure.com/providers/Microsoft.Cdn/operations?api-version=2022-01-01-preview [404 Not Found, 929B, 19ms]` . It's also shown in API Readiness Check log.

**Comment by priyadmurugan** [comment]
> GET https://management.azure.com/providers/Microsoft.Cdn/operations?api-version=2022-01-01-preview [404 Not Found, 929B, 19ms]

Is it possible to change the endpoint to point to staging environment where this new version will be available sooner? We would like to have this swagger changes merged ASAP, but with ARM SDP, it might take a few weeks for the new version to be available globally. Could you suggest any alternative that we can use. Thanks

**Comment by leni-msft** [comment]
> We would like to have this swagger changes merged ASAP
May I know the purpose?

**Comment by priyadmurugan** [comment]
> > We would like to have this swagger changes merged ASAP
> 
> May I know the purpose?

new API version is only for internal purposes and will not be used by external customers. Also, we are planning to release this version only to stage environment in RP (cdn) and the api version is currently present only in Brazil US ARM region.  So if we have an alternative to satisfy this check, it would be great.

**Pull request closed without merging**